CRM/Contribute/Form/Task/PDFLetterCommon.php

```diff
@@ -263,6 +263,11 @@ public static function buildContributionArray($groupBy, $contributionIDs, $retur
         'id' => $contributionId,
         'return' => $returnValues,
       ));
+      foreach ($contribution as $token => &$value) {
+        if ($cfID = CRM_Core_BAO_CustomField::getKeyID($token)) {
```

Comment:
This construct tends to be pretty controversial - mostly because IDEs don't like it. When the assignment is required I think https://stackoverflow.com/questions/630035/zend-studio-reports-warning-assignment-in-condition-is-this-so-bad/2576969#2576969 works - but in this case you can just drop the $cfID

Reply:
@eileenmcnaughton got rid of unused variable